src/PSRule.Rules.Azure/rules/Azure.AppConfig.Rule.ps1

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+#
+# Validation rules for App Configuration
+#
+
+#region Rules
+
+# Synopsis: Ensure app configuration stores audit diagnostic logs are enabled.
+Rule 'Azure.AppConfig.AuditLogs' -Ref 'AZR-000311' -Type 'Microsoft.AppConfiguration/configurationStores' -Tag @{ release = 'GA'; ruleSet = '2022_09' } {
+    $diagnosticLogs = @(GetSubResources -ResourceType 'Microsoft.Insights/diagnosticSettings')
+
+    $Assert.Greater($diagnosticLogs, '.', 0).Reason($LocalizedData.DiagnosticSettingsNotConfigured)
+
+    $logCategoryGroups = 'audit', 'allLogs'
+    $joinedLogCategoryGroups = $logCategoryGroups -join ', '
+
+    foreach ($setting in $diagnosticLogs) {
+        $path = $setting._PSRule.path;
+        $auditLogs = @($setting.Properties.logs | Where-Object {
+            ($_.category -eq 'Audit' -or $_.categoryGroup -in $logCategoryGroups) -and $_.enabled
+        })
+
+        $Assert.Greater($auditLogs, '.', 0).ReasonFrom(
+            'properties.logs',
+            $LocalizedData.AutomationAccountDiagnosticSetting,
+            $setting.name,
+            'Audit',
+            $joinedLogCategoryGroups
+        ).PathPrefix($path)
+    }
```

Comment:
@bengeset96 Great start. Although realizing now there is a bug in `Azure.Automation.AuditLogs`. Which is if multiple diagnostic log settings are configured they all must have auditing enabled, which is not really a requirement.

We want to get to, auditing either through category or category group being configured at least once.

Have a look at https://github.com/Azure/PSRule.Rules.Azure/blob/main/src/PSRule.Rules.Azure/rules/Azure.KeyVault.Rule.ps1#L27-L34 which is much simpler but doesn't handle category groups.

Ultimately we want a combination of both, for this rule and we can use the final code to improve the logic of the other two rules.

Reply:
@BernieWhite

Yes, actually it looks like one resource can support up to _five(5)_ different diagnostic settings. 

I'm in the tinkering process right now. 

```$logCategoryGroups = 'audit', 'allLogs'
    $auditDiagnosticLogs = @(GetSubResources -ResourceType 'Microsoft.Insights/diagnosticSettings' | ForEach-Object {
            $_.Properties.logs | Where-Object { ($_.category -eq 'Audit' -or $_.categoryGroup -in $logCategoryGroups) -and $_.enabled
        }    
    })
   ```
 If doing so, I'd loose the opportunity with having `ReasonFrom()` for each item in the collection with diagnostic name and `PathPrefix()` for each item in the collection. That is a real bummer for the end-user.
 
If I use the some kind of assertion method inside `foreach` or `ForEach-Object` like previously, I'd end up with the exact same problem as we're dealing with now.
 
Without overengineering it, do you see any feasable solution that comes to your mind? Some kind of looping? When one item in the collection returns "true" for audit configured, then write it to to a variable. Then all items in general is checking for this variable and if count "1" is met on the variable, then Pass() and if not, the ReasonFrom method with details about prefix object path, diagnostics name etc. But still the problem will be, if the collection retuns "true" in the middle or late in the collection, then already other items have written to the ReasonFrom method..

Reply:
@BernieWhite Agreed, this makes sense.